src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataArchiveValidationService.cs

```diff
@@ -18,99 +23,274 @@ namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
     public class DataArchiveValidationService : IDataArchiveValidationService
     {
         private readonly IFileTypeService _fileTypeService;
+        private readonly IFileUploadsValidatorService _fileUploadsValidatorService;
 
-        private const int MaxFilenameSize = 150;
+        public const int MaxFilenameSize = 150;
 
-        private static readonly Dictionary<FileType, IEnumerable<Regex>> AllowedMimeTypesByFileType =
-            new()
-            {
-                {DataZip, AllowedArchiveMimeTypes}
-            };
-
-        public DataArchiveValidationService(IFileTypeService fileTypeService)
+        public DataArchiveValidationService(
+            IFileTypeService fileTypeService,
+            IFileUploadsValidatorService fileUploadsValidatorService)
         {
             _fileTypeService = fileTypeService;
+            _fileUploadsValidatorService = fileUploadsValidatorService;
         }
```

Comment:
Done.

I've not been converting constructors to use a primary constructor because I think it pollutes the PR quite a bit. I'd prefer to make the life of the reviewer easier over switching the constructor.

Maybe we can have someone just switch to primary constructors across the codebase in it's own ticket and get it done with - can't imagine it would take long and make PRs much easier to review than when such changes are intermingled. I'll create a ticket.